catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -0,0 +1,236 @@
+      response.status(HttpStatus.BAD_REQUEST_400);
+      return ImmutableMap.of("message", "Bad request");
+    }
+
+    Map<String, Serializable> arguments;
+
+    try {
+      arguments = mapper.readValue(body, Arguments.class).getSerializableArguments();
+    } catch (NullPointerException e) {
+      arguments = Collections.emptyMap();
+    }
+
+    LOGGER.trace("Finding transformer to transform query response.");
+
+    ServiceReference<QueryResponseTransformer> queryResponseTransformer =
+        queryResponseTransformers
+            .stream()
+            .filter(transformer -> transformerId.equals(transformer.getProperty("id")))
+            .findFirst()
+            .orElse(null);
+
+    if (queryResponseTransformer == null) {
+      LOGGER.debug("Could not find transformer with id: {}", transformerId);
+      response.status(HttpStatus.NOT_FOUND_404);
+      return ImmutableMap.of("message", "Service not found");
+    }
+
+    CqlQueryResponse cqlQueryResponse = util.executeCqlQuery(cqlRequest);
+
+    attachFileToResponse(request, response, queryResponseTransformer, cqlQueryResponse, arguments);
+
+    return "";
+  }
+
+  private void setHttpHeaders(Request request, Response response, BinaryContent content)
+      throws MimeTypeException, IllegalArgumentException, NullPointerException {
+    String mimeType = content.getMimeTypeValue();
+
+    if (mimeType == null) {
+      LOGGER.debug("Failure to fetch file extension, mime-type is empty");
+      throw new MimeTypeException("Binary Content contains null mime-type value.");
+    }
+
+    String fileExt = getFileExtFromMimeType(mimeType);
+
+    if (fileExt == null) {
+      LOGGER.debug("Invalid file extension from mimetype {}", mimeType);
+      throw new NullPointerException("Mime-type resulted in null file extension");
+    }
+
+    if (containsGzip(request)) {
+      LOGGER.trace("Request header accepts gzip");
+      response.header(HttpHeaders.CONTENT_ENCODING, GZIP);
+    }
+
+    response.type(mimeType);
+    String attachment =
+        String.format("attachment;filename=export-%s%s", Instant.now().toString(), fileExt);
+    response.header(HttpHeaders.CONTENT_DISPOSITION, attachment);
+  }
+
+  private String getFileExtFromMimeType(String mimeType)
+      throws MimeTypeException, IllegalArgumentException {
+    MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+    String fileExt = allTypes.forName(mimeType).getExtension();
+    if (fileExt == null || StringUtils.isEmpty(fileExt)) {
+      LOGGER.debug("Fetching file extension from mime-type resulted in null or empty.");
+      throw new IllegalArgumentException("Failure fetching file extension from mime type");
+    }
+    return fileExt;
+  }
+
+  private Boolean containsGzip(Request request) throws NullPointerException {
+    Boolean containsGzip = false;
+    try {
+      containsGzip = request.headers(HttpHeaders.ACCEPT_ENCODING).toLowerCase().contains(GZIP);
+    } catch (NullPointerException e) {
```

Comment:
why are we catching an npe here?

Reply:
`request.headers("something-invalid").toLowerCase().contains(GZIP)` throws NPE

Reply:
should be calling `request.headers("something-invalid")` and checking if isn't null instead of catching an npe